File: praktikum1/BinaereSuche.py
def binareSearch(a, target, begin, end):

    if begin <= end :
        mitte = begin + (end - begin) // 2


        if a[mitte] == target: 
            return mitte
        elif a[mitte] > target: 
            return binareSearch(a, target, begin, mitte - 1)
        else :
            return binareSearch(a, target, mitte + 1, end)
    return -1


a = [99, 0, 33, 27, 87, 80, 65, 77, 39, 88]

print(a)
a.sort()
print(a)
target = int(input("Enter the target number: ")) 
position = binareSearch(a, target, 0, len(a)-1)

print(target)

if position != -1: 
    print("Die Position der gescuhten Zahl ist ",  position)
else:
    print( "Zahl nicht gefunden.")

File: praktikum1/BinaereSuche.java
package praktikum1;

import javax.swing.JOptionPane;

public class BinaereSuche {
    public static void main(String[] args) {

        int[] array = {99, 90, 79, 75, 70, 60, 55, 37, 23, 17, 16, 9, 8, 6, 5, 1};

        System.out.println("Array:");
        for (int num : array) {
            System.out.print(num + " ");
        }
        System.out.println();

        String eingabe = JOptionPane.showInputDialog("Geben Sie die gesuchte Zahl ein.");

        int ziel = Integer.parseInt(eingabe);
        int position = binaereSuche(array, ziel, 0, array.length - 1);

        if (position != -1) {
            JOptionPane.showMessageDialog(null,"Die Position der gescuhten Zahl ist " + position);
        } else {
            JOptionPane.showMessageDialog(null, "Zahl nicht gefunden.");
        }
    }

    public static int binaereSuche(int[] array, int ziel, int links, int rechts) {
        if (links <= rechts) {

            int mitte = links + (rechts - links) / 2;
            System.out.println("Überprüfe Mitte: " + mitte);

            if (array[mitte] == ziel) return mitte;
            if (array[mitte] > ziel) return binaereSuche(array, ziel, mitte + 1, rechts);

            return binaereSuche(array, ziel, links, mitte - 1);
        }
        return -1;
    }
}